File: japimsdc/MsdcQDatAPar.java
/**
  * \file MsdcQDatAPar.h
  * Java API code for record of table TblMsdcQDatAPar
  * \author Alexander Wirthmueller
  * \date created: 15 Aug 2018
  * \date modified: 15 Aug 2018
  */

package apimsdc;

import org.w3c.dom.*;
import sbecore.*;

public class MsdcQDatAPar {
	
	public MsdcQDatAPar(
				int jnum
				, String ftmX1Start
				, String x2SrefKKey
				, String titX2SrefKKey
				, String stubRefMsdcMFile
				, String Val
				, String Bindata
			) {
		this.jnum = jnum;
		this.ftmX1Start = ftmX1Start;
		this.x2SrefKKey = x2SrefKKey;
		this.titX2SrefKKey = titX2SrefKKey;
		this.stubRefMsdcMFile = stubRefMsdcMFile;
		this.Val = Val;
		this.Bindata = Bindata;
	};

	public int jnum;
	public String ftmX1Start;
	public String x2SrefKKey;
	public String titX2SrefKKey;
	public String stubRefMsdcMFile;
	public String Val;
	public String Bindata;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "MsdcQDatAPar");

		if (Xmlio.checkXPath(doc, basexpath)) {
			ftmX1Start = Xmlio.extractStringUclc(doc, basexpath, "ftmX1Start", "sta", null, 0);
			x2SrefKKey = Xmlio.extractStringUclc(doc, basexpath, "x2SrefKKey", "key", null, 0);
			titX2SrefKKey = Xmlio.extractStringUclc(doc, basexpath, "titX2SrefKKey", "key2", null, 0);
			stubRefMsdcMFile = Xmlio.extractStringUclc(doc, basexpath, "stubRefMsdcMFile", "fil", null, 0);
			Val = Xmlio.extractStringUclc(doc, basexpath, "Val", "val", null, 0);
			Bindata = Xmlio.extractStringUclc(doc, basexpath, "Bindata", "bin", null, 0);
			
			return true;
		};

		return false;
	};
};

File: apimsdc/MsdcQDatAPar.cpp
/**
  * \file MsdcQDatAPar.cpp
  * API code for table TblMsdcQDatAPar (implementation)
  * \author Alexander Wirthmueller
  * \date created: 4 Oct 2018
  * \date modified: 4 Oct 2018
  */

#ifdef _WIN32
	#include "stdafx.h"
#endif

#include "MsdcQDatAPar.h"

/******************************************************************************
 class MsdcQDatAPar
 ******************************************************************************/

MsdcQDatAPar::MsdcQDatAPar(
			const uint jnum
			, const string ftmX1Start
			, const string x2SrefKKey
			, const string titX2SrefKKey
			, const string stubRefMsdcMFile
			, const string Val
			, const string Bindata
		) {
	this->jnum = jnum;
	this->ftmX1Start = ftmX1Start;
	this->x2SrefKKey = x2SrefKKey;
	this->titX2SrefKKey = titX2SrefKKey;
	this->stubRefMsdcMFile = stubRefMsdcMFile;
	this->Val = Val;
	this->Bindata = Bindata;
};

bool MsdcQDatAPar::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "MsdcQDatAPar");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "ftmX1Start", "sta", ftmX1Start);
		extractStringUclc(docctx, basexpath, "x2SrefKKey", "key", x2SrefKKey);
		extractStringUclc(docctx, basexpath, "titX2SrefKKey", "key2", titX2SrefKKey);
		extractStringUclc(docctx, basexpath, "stubRefMsdcMFile", "fil", stubRefMsdcMFile);
		extractStringUclc(docctx, basexpath, "Val", "val", Val);
		extractStringUclc(docctx, basexpath, "Bindata", "bin", Bindata);
	};

	return basefound;
};

/******************************************************************************
 class ListMsdcQDatAPar
 ******************************************************************************/

ListMsdcQDatAPar::ListMsdcQDatAPar() {
};

ListMsdcQDatAPar::ListMsdcQDatAPar(
			const ListMsdcQDatAPar& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i=0;i<nodes.size();i++) nodes[i] = new MsdcQDatAPar(*(src.nodes[i]));
};

ListMsdcQDatAPar& ListMsdcQDatAPar::operator=(
			const ListMsdcQDatAPar& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i=0;i<nodes.size();i++) nodes[i] = new MsdcQDatAPar(*(src.nodes[i]));

	return *this;
};

ListMsdcQDatAPar::~ListMsdcQDatAPar() {
	clear();
};

void ListMsdcQDatAPar::clear() {
	for (unsigned int i=0;i<nodes.size();i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListMsdcQDatAPar::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	MsdcQDatAPar* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListMsdcQDatAPar");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "MsdcQDatAPar", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i=0;i<ics.size();i++) {
		rec = new MsdcQDatAPar(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/MsdcQDatAPar[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

